src/jsonrpc.rs

```diff
@@ -0,0 +1,25 @@
+use jsonrpc_stdio_server::{ServerBuilder, jsonrpc_core::*};
+use tracing::{info, trace};
+
+pub async fn start_server() {
+    info!("Starting jsonrpc server");
+    let server = new_server();
+    let server = server.build();
+    server.await;
+}
+
+pub fn new_server() -> ServerBuilder {
+    let mut io = IoHandler::default();
+    handle_ping(&mut io);
+    let request = r#"{"jsonrpc": "2.0", "method": "ping", "params": {}, "id": 1}"#;
+	let response = r#"{"jsonrpc":"2.0","result":"pong","id":1}"#;
+    assert_eq!(io.handle_request_sync(request), Some(response.to_string()));
+    return ServerBuilder::new(io)
+}
+
+fn handle_ping(handler:&mut IoHandler) {
+    handler.add_sync_method("ping", |_params|{
```

Comment:
Have you tried the `derive` way? It seems cool :sunglasses: 

```
#[rpc]
pub trait Rpc {
	/// Adds two numbers and returns a result
	#[rpc(name = "add")]
	fn add(&self, u64, u64) -> Result<u64>;
}

pub struct RpcImpl;
impl Rpc for RpcImpl {
	fn add(&self, a: u64, b: u64) -> Result<u64> {
		Ok(a + b)
	}
}
```

Reply:
Cool! Will definitely switch to it.